pandas/core/internals/managers.py

```diff
@@ -1704,23 +1709,25 @@ def construction_error(tot_items, block_shape, axes, e=None):
     if len(implied) <= 2:
         implied = implied[::-1]
 
+    # We return the exception object instead of raising it so that we
+    #  can raise it in the caller; mypy plays better with that
     if passed == implied and e is not None:
-        raise e
+        return e
```

Comment:
Rather than do this I think would be better to just mark this function with `typing.NoReturn` as the return value

Reply:
i recall that fails on our min version build.

Reply:
elsewhere (in particular in indexes.base) we've moved towards this pattern